docs/03-pages/01-building-your-application/03-data-fetching/03-get-server-side-props.mdx

```diff
@@ -3,77 +3,85 @@ title: getServerSideProps
 description: Fetch data on each request with `getServerSideProps`.
 ---
 
-If you export a function called `getServerSideProps` (Server-Side Rendering) from a page, Next.js will pre-render this page on each request using the data returned by `getServerSideProps`.
+`getServerSideProps` is a Next.js function that can be used fetch data and render the contents of a page at request time.
+
+## Example
+
+You can use `getServerSideProps` by exporting it from a Page Component:
 
 ```tsx filename="pages/index.tsx" switcher
 import type { InferGetServerSidePropsType, GetServerSideProps } from 'next'
 
-type Repo = {
-  name: string
-  stargazers_count: number
+type Data = {
+  title: string
+  description: string
+}
+
+type Props = InferGetServerSidePropsType<typeof getServerSideProps>
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  const res = await fetch(`https://.../data`)
+  const data: Data = await res.json()
+
+  // Pass data to the page via props
+  return { props: { data } }
 }
 
-export const getServerSideProps = (async (context) => {
-  const res = await fetch('https://api.github.com/repos/vercel/next.js')
```

Comment:
The only nice thing about this is that you can copy paste it and it works – should we consider using `api.vercel.app`?

Reply:
Sure! Updated 🙏🏼